go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +52,429 @@ func TestExpansionCosts(t *testing.T) {
+			offset:   9,
+			expected: bytes.Repeat([]byte{0}, len(baseTarget))},
+		"padding needed": {
+			target:   baseTarget,
+			store:    baseStore,
+			offset:   4,
+			expected: []byte{0x90, 0xab, 0xcd, 0xef, 0, 0, 0}},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			m := NewMemory()
+			m.store = test.store
+			target := test.target
+
+			m.CopyData(test.offset, target)
+			if !bytes.Equal(target, test.expected) {
+				t.Errorf("unexpected target value, want: %x, got: %x", test.expected, target)
+			}
+		})
+	}
+
+}
+
+func TestMemory_GetSliceWithCapacity(t *testing.T) {
+
+	baseStore := []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+	memorySize := uint64(8)
+	baseEmpty := bytes.Repeat([]byte{0}, int(memorySize))
+
+	tests := map[string]struct {
+		store    []byte
+		offset   uint64
+		size     uint64
+		expected []byte
+	}{
+		"regular": {
+			store:    baseStore,
+			offset:   0,
+			size:     memorySize,
+			expected: baseStore},
+		"regular in empty memory": {
+			store:    []byte{},
+			offset:   0,
+			size:     memorySize,
+			expected: baseEmpty},
+		"size zero": {
+			store:  baseStore,
+			offset: 0,
+			size:   0},
+		"expand memory": {
+			store:    baseStore,
+			offset:   0,
+			size:     memorySize + 4,
+			expected: append(baseStore, baseEmpty[:4]...)},
+		"offset bigger than memory": {
+			store:  baseStore,
+			offset: uint64(len(baseStore) + 2),
+			size:   4},
+		"size+offset bigger than memory": {
+			store:  baseStore,
+			offset: 4,
+			size:   8},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			m := NewMemory()
+			m.store = test.store
+
+			result := m.GetSliceWithCapacity(test.offset, test.size)
+			if !bytes.Equal(result, test.expected) {
+				t.Errorf("unexpected result, want: %x, got: %x", test.expected, result)
+			}
+		})
+	}
+}
+
+func TestMemory_Print(t *testing.T) {
+
+	tests := map[string]struct {
+		store    []byte
+		expected string
+	}{
+		"empty": {store: []byte{},
+			expected: "### mem 0 bytes ###\n-- empty --\n####################\n"},
+		"8 bytes": {store: []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef},
+			expected: "### mem 8 bytes ###\n000: 12 34 56 78 90 ab cd ef\n####################\n"},
+		"32 bytes": {store: []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef,
+			0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef,
+			0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef,
+			0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef},
+			expected: "### mem 32 bytes ###\n000: 12 34 56 78 90 ab cd ef 12 34 56 78 90 ab cd ef 12 34 56 78 90 ab cd ef 12 34 56 78 90 ab cd ef\n####################\n"},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			m := NewMemory()
+			m.store = test.store
+
+			// redirect stdout
+			old := os.Stdout
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+
+			// Run testing code
+			m.Print()
+			// read the output
+			w.Close()
+			out, _ := io.ReadAll(r)
+			os.Stdout = old
```

Comment:
Please update the "Print()" function to return a string instead of writing stuff to `Stdout`.